InMobi/src/main/java/com/mopub/mobileads/InMobiInterstitial.kt

```diff
@@ -0,0 +1,171 @@
+
+import android.app.Activity
+import android.content.Context
+import com.inmobi.ads.AdMetaInfo
+import com.inmobi.ads.InMobiAdRequestStatus
+import com.inmobi.ads.InMobiInterstitial
+import com.inmobi.ads.exceptions.SdkNotInitializedException
+import com.inmobi.ads.listeners.InterstitialAdEventListener
+import com.inmobi.sdk.InMobiSdk
+import com.mopub.common.DataKeys
+import com.mopub.common.LifecycleListener
+import com.mopub.common.logging.MoPubLog
+import com.mopub.common.logging.MoPubLog.AdapterLogEvent
+import com.mopub.mobileads.InMobiAdapterConfiguration.Companion.onInMobiAdFailWithError
+import com.mopub.mobileads.InMobiAdapterConfiguration.Companion.onInMobiAdFailWithEvent
+import java.lang.Error
+
+open class InMobiInterstitial : BaseAd() {
+
+    companion object {
+        val ADAPTER_NAME: String = InMobiInterstitial::class.java.simpleName
+    }
+
+    private var mPlacementId: Long? = null
+    private var mInMobiInterstitial: InMobiInterstitial? = null
+
+    override fun onInvalidate() {
+        MoPubLog.log(AdapterLogEvent.CUSTOM, ADAPTER_NAME, "InMobi interstitial destroyed")
+        if (mInMobiInterstitial != null) {
+            mInMobiInterstitial = null
+        }
+    }
+
+    override fun getLifecycleListener(): LifecycleListener? {
+        return null
+    }
+
+    override fun getAdNetworkId(): String {
+        return mPlacementId?.toString() ?: ""
+    }
+
+    override fun checkAndInitializeSdk(launcherActivity: Activity, adData: AdData): Boolean {
+        return false
+    }
+
+    override fun load(context: Context, adData: AdData) {
+        setAutomaticImpressionAndClickTracking(false)
+        val extras: Map<String, String> = adData.extras
+
+        if (InMobiAdapterConfiguration.isInMobiSdkInitialised) {
+            loadInterstitial(context, adData, extras)
+        } else {
+            InMobiAdapterConfiguration.initialiseInMobi(extras, context, object : InMobiAdapterConfiguration.InitCompletionListener {
+                override fun onSuccess() {
+                    loadInterstitial(context, adData, extras)
+                }
+
+                override fun onFailure(error: Error?, exception: Exception?) {
+                    error?.let {
+                        onInMobiAdFailWithEvent(AdapterLogEvent.LOAD_FAILED, adNetworkId, MoPubErrorCode.ADAPTER_CONFIGURATION_ERROR,
+                                "InMobi banner request failed due to InMobi initialization failed with a reason: ${error.message}",
+                                com.mopub.mobileads.InMobiBanner.ADAPTER_NAME, mLoadListener, null)
+                    }
+
+                    exception?.let {
+                        onInMobiAdFailWithError(it, MoPubErrorCode.ADAPTER_CONFIGURATION_ERROR,
+                                "InMobi banner request failed due to InMobi initialization failed with an exception.",
+                                InMobiBanner.ADAPTER_NAME, mLoadListener, null)
+                    }
+                }
```

Comment:
What if both `error` and `exception` are not null, it will end up giving load fail callback multiple times.
Right now only one will be returned but this is still something which I feel is not right because one is dependent on other.